### PR TITLE
Refactor library into sub-libraries eventuals-events and eventuals-http and only build eventuals-http on macOS and Linux

### PR DESCRIPTION
**Pull request by benh**
Addresses #60.

### CODE REVIEW

File: BUILD.bazel [hunk 1]
```diff
@@ -30,27 +30,64 @@ cc_library(
...
     ],
     srcs = [
         "stout/scheduler.cc",
         "stout/static-thread-pool.cc",
-        "stout/event-loop.cc",
     ],
     deps = [
         "@com_github_google_glog//:glog",
+    ],
+    visibility = ["//visibility:public"],
```
Comment (FolMing) [inline BUILD.bazel:47]:
```suggestion
    visibility = ["//visibility:private"],
```

File: BUILD.bazel [hunk 2]
```diff
@@ -30,27 +30,64 @@ cc_library(
...
+        "stout/signal.h",
+    ],
+    srcs = [
+        "stout/event-loop.cc",
+    ],
+    deps = [
+        ":eventuals-base",
         "@com_github_libuv_libuv//:libuv",
+    ],
+    visibility = ["//visibility:public"],
```
Comment (FolMing) [inline BUILD.bazel:66]:
```suggestion
    visibility = ["//visibility:private"],
```

File: BUILD.bazel [hunk 3]
```diff
@@ -30,27 +30,64 @@ cc_library(
...
+)
+
+cc_library(
+    name = "eventuals-http",
+    deps = [
+        ":eventuals-base",
         "@com_github_curl_curl//:libcurl",
     ],
     visibility = ["//visibility:public"],
```
Comment (FolMing) [inline BUILD.bazel:75]:
```suggestion
    visibility = ["//visibility:private"],
```

**Review by FolMing — COMMENTED** [review]
We shouldn't allow users depend on sub-libraries, or should we?

File: BUILD.bazel (hunk 1 above)
Comment (benh) [reply BUILD.bazel]: Sounds good, I'll make the sub-libraries private!

**Review by FolMing — APPROVED** [review]

**Pull request merged**